### PR TITLE
Add: custom js visualizations

### PR DESCRIPTION
**Pull request by luke14free**
test

### CODE REVIEW

File: frontend/app/visualizations/chart/index.js [hunk 1]
```diff
@@ -75,6 +76,11 @@ function ChartEditor(ColorPalette) {
         });
       };
 
+      scope.options.customCode = "// Available variables are x, ys, element, and Plotly.\n";
```
Comment (arikfr) [inline frontend/app/visualizations/chart/index.js]: Please use ES6 template strings here (created with backticks - "``"), which supports multi line strings.

File: frontend/package.json [hunk 2]
```diff
@@ -53,7 +53,7 @@
     "ng-annotate": "^1.2.1",
     "ng-annotate-loader": "^0.2.0",
     "ng-table": "^2.1.0",
-    "plotly.js": "^1.16.0",
+    "plotly.js": "^1.2.0",
```
Comment (arikfr) [inline frontend/package.json]: ?

**Comment by luke14free** [comment]
@arikfr I added the value to the clientConfig, but it was somehow defined as a constant in the `index.js` file. I guess that's it's linked to the fact that you want to separate backend and frontend and are still mocking that stuff, so I also added temporarily the switch to that constant in order to test the feature.

File: frontend/app/visualizations/chart/index.js [hunk 3]
```diff
@@ -66,6 +65,11 @@ function ChartEditor(ColorPalette) {
         scatter: { name: 'Scatter', icon: 'circle-o' },
       };
 
+      console.log(clientConfig);
+      if (clientConfig.allowCustomJSVisualizations) {
```
Comment (arikfr) [inline frontend/app/visualizations/chart/index.js]: Need to use this in the render function too, so that someone clever won't manually submit this type.

**Comment by arikfr** [comment]
@luke14free you understood it right about the `clientConfig` 👍

**Comment by agenorfbp** [comment]
in my redash don't have many files listed in this pull, example: index.js.
I need create those files ?

**Comment by agenorfbp** [comment]
This customization is very important for me and my organization, this Should be including in version 1.0

**Pull request merged**